src/ApplicationLogs/Store/Models/BlockchainEventModel.cs

```diff
@@ -12,15 +12,14 @@
 using ApplicationLogs.Store.States;
 using Neo;
 using Neo.VM.Types;
-using Array = System.Array;
 
 namespace ApplicationLogs.Store.Models
 {
     public class BlockchainEventModel
     {
-        public UInt160 ScriptHash { get; private init; } = UInt160.Zero;
+        public UInt160 ScriptHash { get; private init; } = new();
```

Comment:
Why create a new empty instance if we have one in memory?

Reply:
Its a memory leak. That's the reason why we had this problem with different addresses in the state. It uses the `class` instance of `UInt160.Zero` when doing `ScriptHash.Deserialize()`. It will keep writing to the same buffer. So i need a new instance either way.

Reply:
If we reuse the same instance, how can be a memory leak?

Reply:
@shargon we are using the same instance. That was the problem. The instance being `UInt160.Zero`